src/plugins/intel_gpu/src/graph/reduce.cpp

```diff
@@ -95,6 +97,126 @@ layout reduce_inst::calc_output_layout(reduce_node const& node, kernel_impl_para
         return layout{output_type, input_format, tensor(batch(in_dims[0]), feature(in_dims[1]), spatial(in_dims[2], in_dims[3]))};
 }
 
+template<typename ShapeType>
+std::vector<layout> reduce_inst::calc_output_layouts(reduce_node const& /*node*/, kernel_impl_params const& impl_param) {
+    auto desc = impl_param.typed_desc<reduce>();
+
+    auto input0_layout = impl_param.get_input_layout(0);
+
+    // get 'output_shapes' by shape_infer of ngraph
+
+    // input_shapes
+    std::vector<ShapeType> input_shapes = {
+        input0_layout.get<ShapeType>(),
+        ShapeType{0}
+    };
+
+    // output_shapes
```

Comment:
nit: some comments (like this one) are obvious and pretty much useless.

Reply:
Done